src/views/Leads/GridItem/index.js

```diff
@@ -0,0 +1,298 @@
+                        source: lead.source,
+                        confidentiality: lead.confidentiality,
+                        assignee: lead.assignee,
+                        publishedOn: lead.publishedOn,
+                        attachment: lead.attachment,
+                        website: lead.website,
+                        leadGroup: lead.leadGroup,
+                        url: lead.url,
+                        text: lead.text,
+                    },
+                },
+            },
+            editEntries: reverseRoute(
+                pathNames.editEntries,
+                {
+                    projectId: activeProject,
+                    leadId: lead.id,
+                },
+            ),
+        };
+    }
+
+    get mimeIcon() {
+        const { lead } = this.props;
+        let icon = iconNames.documentText;
+        if (lead.attachment) {
+            icon = leadTypeIconMap[lead.attachment.mimeType];
+        } else if (lead.url) {
+            icon = iconNames.globe;
+        }
+        return icon;
+    }
+
+    getThumbnail = () => {
+        const { lead } = this.props;
+
+        let thumbnail = `url(${leadThumbnail})`;
+
+        if (lead.thumbnail) {
+            thumbnail = `url(${lead.thumbnail})`;
+        } else if (lead.attachment &&
+            leadPaneTypeMap[lead.attachment.mimeType] === LEAD_PANE_TYPE.image) {
+            thumbnail = `url(${lead.attachment.file})`;
+        }
+
+        return thumbnail;
+    }
+
+    renderMarkAction = () => {
+        const { lead, onMarkPending, onMarkProcessed } = this.props;
+
+        if (lead.status === 'pending') {
+            return (
+
+                <Button
+                    tabIndex="-1"
+                    title={_ts('leads', 'markAsProcessedTitle')}
+                    className={classNames(styles.markProcessed, styles.mark)}
+                    onClick={() => onMarkProcessed(lead)}
+                    transparent
+                    iconName={iconNames.check}
+                />
+            );
+        } else if (lead.status === 'processed') {
+            return (
+                <Button
+                    tabIndex="-1"
+                    title={_ts('leads', 'markAsPendingTitle')}
+                    className={classNames(styles.markPending, styles.mark)}
+                    onClick={() => onMarkPending(lead)}
+                    transparent
+                    iconName={iconNames.close}
+                />
+            );
+        }
+
+        return null;
+    }
+
+    renderActions = () => {
+        const { lead, onSearchSimilarLead, onRemoveLead } = this.props;
+        const MarkAction = this.renderMarkAction;
+
+        return (
+            <React.Fragment>
+                <div className={styles.mainActions}>
+                    <Cloak
+                        hide={({ hasAnalysisFramework }) => !hasAnalysisFramework}
+                        render={({ disabled }) => (
```

Comment:
The API for Cloak has been changed.
Instead of a class, render prop takes component type.